### ISSUE TITLE
Add documentation about GitHub release

### ISSUE THREAD

**Issue by stefandesu** [gbv/cocoda#415]
Starting with 0.9.2, Travis automatically creates a zip release of the Cocoda build for every new version. We should add documentation on how to use this and how to adjust the configuration.

Continuation of #169.

**Comment by nichtich** [comment]
The releases should also be used for deployment at http://coli-conc.gbv.de/cocoda/app/ so we can make sure they actually work.

**Comment by stefandesu** [comment]
> The releases should also be used for deployment at http://coli-conc.gbv.de/cocoda/app/ so we can make sure they actually work.

Right now, I'm not actually sure if this is a good idea. Right now, we have a uniform way of updating instances on our server (`./update <name>`). If you look at how Travis is creating the asset for the release, it takes the build created for GitHub Pages and zips it. So as long as the instance on Github Pages is working, the release build should work as well.

Alternatively, we could update the release instance using a webhook for a published release. On our server, the webhook handler would run the commands just like specified in the documentation.

**Comment by nichtich** [comment]
> Alternatively, we could update the release instance using a webhook for a published release. On our server, the webhook handler would run the commands just like specified in the documentation.

Yes, this would make sure the release artifact actually lead to a working product. Right now this works:

~~~bash
wget https://github.com/gbv/cocoda/releases/download/0.9.2/cocoda-0.9.2.zip
unzip cocoda-0.9.2.zip
cd cocoda
php -S 127.0.0.1:8000 # see https://gist.github.com/willurd/5720255 for alternatives
~~~

but configuring the instance is difficult because the default config is baked into the app. We should think about moving configuration to `cocoda.json`.

An alternative would be to add an additional hook to install each release in its own directory, e.g. https://coli-conc.gbv.de/cocoda/0.9.2/

**Comment by stefandesu** [comment]
> but configuring the instance is difficult because the default config is baked into the app. We should think about moving configuration to `cocoda.json`.

I don't understand this point. There is a `cocoda.json` in the root folder which can be fully used for configuring the instance, including overriding any of the options that are already in the default config. 

> An alternative would be to add an additional hook to install each release in its own directory, e.g. https://coli-conc.gbv.de/cocoda/0.9.2/

I like this idea actually. Only problem is that we'd need to start properly versioning `jskos-server` as well and maybe even configure which version to use in Cocoda so that it continues to work. The current jskos-server should be mostly backward compatible with previous releases though, so maybe it's not a problem.

**Comment by nichtich** [comment]
> I don't understand this point. There is a cocoda.json in the root folder which can be fully used for configuring the instance, including overriding any of the options that are already in the default config

The release does not include the default config file in JSON format to compare values. Maybe use an empty default config for releases and move all default config to `config.json`?

> Only problem is that we'd need to start properly versioning `jskos-server` as well and maybe even configure which version to use in Cocoda so that it continues to work

:+1: in theory semantic versioning would help out. Cocoda should give an error message if configured with a jskos-server instance that is too old (if we introduce breaking changes in jskos-server).

**Comment by stefandesu** [comment]
> The release does not include the default config file in JSON format to compare values. Maybe use an empty default config for releases and move all default config to `config.json`?

To be honest, I don't think this is much of an issue. I personally would really keep the default config separate. What we could do is include the default config as a separate file as well, so we'd have `cocoda.json` and `cocoda.default.json` in the root folder.

> 👍 in theory semantic versioning would help out. Cocoda should give an error message if configured with a jskos-server instance that is too old (if we introduce breaking changes in jskos-server).

That error message will only appear with newer versions of Cocoda though because we have to implement it first. 🙈

**Comment by stefandesu** [comment]
1. I adjusted the documentation a little further and included notes about serving the instance.

2. Our server was updated to get information about releases via a webhook, installing the release with the version number as a subfolder (e.g. https://coli-conc.gbv.de/cocoda/0.9.2/), and updating the symlink from `app` to this version. Currently, there are only [`app`](https://coli-conc.gbv.de/cocoda/app/), [`dev`](https://coli-conc.gbv.de/cocoda/dev/), and [`0.9.2`](https://coli-conc.gbv.de/cocoda/0.9.2/), but I will try to add older releases as well.

For now, there won't be any proper versioning of jskos-server or rather the API, but we should definitely start doing that.
